### PR TITLE
feat(pipes-targets): add EventBridge

### PR DESCRIPTION
**Pull request by msambol**
Add EventBridge event bus as a Pipes target.

### CODE REVIEW

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts:13]: `none` is not a correct `IInputTransformation` value, same for the following properties
```suggestion
   * @default - none
```

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts:21]: As far as I can tell, the CDK doesn't really JSDoc string length constraints like Cfn does:

https://github.com/aws/aws-cdk/blob/185f93d40a643b651be9c6668bc2b0772d40fcef/packages/aws-cdk-lib/aws-cloudfront/lib/cache-policy.ts#L20-L25

https://github.com/aws/aws-cdk/blob/185f93d40a643b651be9c6668bc2b0772d40fcef/packages/aws-cdk-lib/aws-cloudfront/lib/cache-policy.ts#L138-L140

```suggestion
   * A free-form string used to decide what fields to expect in the event detail.
```

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts:35]: I would use an `@example` tag here, something like
```suggestion
   * @example
   * // if the URL for the endpoint is https://abcde.veo.endpoints.event.amazonaws.com
   * 'abcde.veo'
```

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts:92]: Should cleaner to refactor the functions to take the `parameters` directly, and iterate over the validators

```ts
function validateDetailType({detailType}: EventBridgeTargetParameters)
```

```suggestion
    for (const validate of [validateDetailType, ...]) validate(parameters);
```

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts:33]: Same goes for the validation RegExp, I don't usually see them, especially if they are fairly obvious. An example is usually clearer too

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts:67]: Should be `^\$(\.[\w/_-]+(\[(\d+|\*)\])*)*$`, but again, I think an `@example` would be clearer

**Review by nmussy — CHANGES_REQUESTED** [review]
The file names should also be camel cased, `s/eventBridge/event-bridge/`

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/eventBridge.ts:117]: Shouldn't all these max length checks be inclusive?
```suggestion
    if (detailType.length < 1 || detailType.length >= 128) {
```

**Comment by msambol** [comment]
@nmussy Updated. Great feedback as always. Thanks!

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts:75]: This should save you a needless `undefined` reassignment
```suggestion
    if (parameters) {
      this.eventBridgeParameters = parameters;
      for (const validate of [validateDetailType, validateEndpointId, validateSource, validateTime]) {
```

File: packages/@aws-cdk/aws-pipes-targets-alpha/README.md
Comment (kaizencc) [inline packages/@aws-cdk/aws-pipes-targets-alpha/README.md:175]: need to add this to the list of targets at the top of the readme

File: packages/@aws-cdk/aws-pipes-targets-alpha/README.md
Comment (kaizencc) [inline packages/@aws-cdk/aws-pipes-targets-alpha/README.md:183]:
```suggestion
const eventBusTarget = new targets.EventBridgeTarget(targetEventBus);
```

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts
Comment (kaizencc) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts:68]:
```suggestion
  public readonly targetArn: string;

```

Comment (kaizencc) [reply packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts:68]: in addition to the spacing since this is a public prop it should have a docstring... shouldn't the linter have picked up on that?

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts
Comment (kaizencc) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts:76]: i haven't seen this pattern before and maybe its cuz i've been away from typescript but cool!

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts
Comment (kaizencc) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts:100]: shouldn't all these `validateXxx` methods be worried about tokens? (and unit tests to check for that scenario)

**Review by kaizencc — DISMISSED** [review]
Hi @msambol sorry for the long delay, have a few comment before my stamp of approval!

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts
Comment (msambol) [reply packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts:68]: @kaizencc I think because it's here ? https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-pipes-alpha/lib/target.ts#L19-L22. not entirely sure.

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts
Comment (msambol) [reply packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts:102]: added

**Comment by msambol** [comment]
@kaizencc – Update per your feedback. Thanks for the review!

**Comment by msambol** [comment]
@kaizencc mind giving this another pass?

**Comment by msambol** [comment]
@kaizencc I updated the integration test to incorporate feedback by @moelasmar on https://github.com/aws/aws-cdk/pull/30756.

**Comment by msambol** [comment]
@Leo10Gama Could we get this one in next?

File: packages/@aws-cdk/aws-pipes-targets-alpha/README.md
Comment (Leo10Gama) [inline packages/@aws-cdk/aws-pipes-targets-alpha/README.md:37]:
```suggestion
6. `targets.EventBridgeTarget`: [Send event source to an EventBridge event bus](#amazon-eventbridge-event-bus)
```

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts
Comment (Leo10Gama) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/event-bridge.ts:64]:
```suggestion
 * An EventBridge Pipes target that sends messages to an EventBridge event bus.
```

File: packages/@aws-cdk/aws-pipes-targets-alpha/test/event-bridge.test.ts
Comment (Leo10Gama) [inline packages/@aws-cdk/aws-pipes-targets-alpha/test/event-bridge.test.ts:197]:
```suggestion
  test('Source must be <= 256 characters', () => {
```

File: packages/@aws-cdk/aws-pipes-targets-alpha/test/event-bridge.test.ts
Comment (Leo10Gama) [inline packages/@aws-cdk/aws-pipes-targets-alpha/test/event-bridge.test.ts:225]:
```suggestion
  test('Time must be <= 256 characters', () => {
```

**Review by Leo10Gama — COMMENTED** [review]
Thanks for the contribution! Aside from a few minor typos, it looks pretty good to me

**Review by Leo10Gama — APPROVED** [review]
LGTM, thanks again!

**Pull request merged**
